tests/setup.py

```diff
@@ -195,3 +197,31 @@ def create_group_request(self, user, group, role="member"):
         group_obj.add_member(
             requester=user_obj, user_or_group=user_obj, reason="", status="pending", role=role
         )
+
+    def create_service_account(self, service_account, owner, description="", machine_set=""):
```

Comment:
This is a really excellent point and is actually something on my radar to fix. What we were aiming for in the initial refactor was to not use any of the legacy code that's going to be deleted, which meant doing a bunch of low-level model work because there were no services or repositories that could do this. However, as soon as there is a service or repository call that can do what `SetupTest` is doing, it absolutely should switch to that call instead of hand-rolling model changes. You just aren't seeing any of that yet because we haven't ported the create and grant flows to usecases yet.

I'll add a big comment to this file to that effect.